nsqd/http.go

```diff
@@ -107,8 +108,7 @@ func freeMemory(w http.ResponseWriter, req *http.Request, ps httprouter.Params)
 
 func (s *httpServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	if !s.tlsEnabled && s.tlsRequired {
-		resp := fmt.Sprintf(`{"message": "TLS_REQUIRED", "https_port": %d}`,
-			s.nsqd.RealHTTPSAddr().Port)
+		resp := fmt.Sprintf(`{"message": "TLS_REQUIRED", "https_addr": "%s"}`, s.nsqd.RealHTTPSAddr())
```

Comment:
This is a backwards-incompatible change, we want to keep the old `https_port` part of the response if nsqd is not using the new unix-socket address feature.

It seems that this may not really be relevant for unix-sockets listeners anyway - you probably wouldn't use tls on the unix socket. Only a local process could connect to a unix socket, tls is less relevant so you'd probably use plain http, or it would be an https proxy like nginx terminating tls then forwarding the request to the nsqd http unix socket. So if tls-required and https-address is a unix socket ... I'd just put 0 for `https_port` for now I guess.

Reply:
Just figured out that `internal/http_api/api_request.go` `GETV1/POSTV1` has retries by that port. I just disable ability to using https via unix socket and revert part of code.